src/promnesia/sources/dogsheep.py

```diff
@@ -0,0 +1,28 @@
+'''
+Uses [[https://github.com/karlicoss/HPI][HPI]] dogsheep module
+'''
+
+from promnesia.common import Visit, Loc, Results
+
+
+def index() -> Results:
+    from . import hpi
+    from my.hackernews import dogsheep
```

Comment:
Thanks, that's good to know. I suppose in order to do this for hackernews, we would have to modify dogsheep and materialistic in HPI to have them both export the same format for each item.

Reply:
Yep, something like this -- it's not always possible, but usually there is a reasonable common denominator (which we put in `common.py`), and then it can be inherited by both specific implementations. 
So if someone cares about merged data they can query it from `.all` submodule, whereas if they'd rather prefer richer data (e.g. API data usually has more fields than GDPR exports), they can use a more specific submodule.